FWCore/Framework/src/EventSetupImpl.cc

```diff
@@ -13,66 +13,44 @@
 //
 
 // system include files
+#include <algorithm>
 
 // user include files
 #include "FWCore/Framework/interface/EventSetupImpl.h"
 #include "FWCore/Framework/interface/EventSetupRecord.h"
-#include "FWCore/Framework/interface/EventSetupKnownRecordsSupplier.h"
+#include "FWCore/Framework/interface/EventSetupRecordKey.h"
+#include "FWCore/Utilities/interface/Exception.h"
 
 namespace edm {
-//
-// constants, enums and typedefs
-//
 
-//
-// static data member definitions
-//
-
-//
-// constructors and destructor
-//
 EventSetupImpl::EventSetupImpl(ActivityRegistry const* activityRegistry) :
-   recordMap_(),
    activityRegistry_(activityRegistry)
-
 {
 }
 
-// EventSetupImpl::EventSetupImpl(EventSetupImpl const& rhs)
-// {
-//    // do actual copying here;
-// }
-
 EventSetupImpl::~EventSetupImpl()
 {
 }
 
-//
-// assignment operators
-//
-// EventSetupImpl const& EventSetupImpl::operator=(EventSetupImpl const& rhs)
-// {
-//   //An exception safe implementation is
-//   EventSetupImpl temp(rhs);
-//   swap(rhs);
-//
-//   return *this;
-// }
-
-//
-// member functions
-//
 void
 EventSetupImpl::insert(const eventsetup::EventSetupRecordKey& iKey,
                 const eventsetup::EventSetupRecordImpl* iRecord)
 {
-   recordMap_[iKey]= iRecord;
+   auto lb = std::lower_bound(keysBegin_, keysEnd_, iKey);
+   if (lb == keysEnd_ || iKey != *lb) {
+      throw cms::Exception("LogicError") << "EventSetupImpl::insert Could not find key\n"
```

Comment:
this error message would be more helpful if it included:
1. information about the key (name, ?)
2. information about what was found instead (another key, or nothing)

Reply:
I believe the exception can only happen if there is a messup in the framework code. As such, if it happens the framework group would have to diagnose the problem anyway.

Reply:
Okay, if I ever encounter this in the future, I will just forward it to you...